content/en/blog/_posts/2022-10-18-kubernetes-1.26-deprecations-and-removals.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: blog
+title: "Kubernetes Removals, Deprecations, and Major Changes in 1.26"
+date: 2022-11-18
+slug: upcoming-changes-in-kubernetes-1-26
+---
+
+**Authors**: Frederico Muñoz
+
+Change is an integral part of the Kubernetes life-cycle: as Kubernetes grows and matures, features may be deprecated, removed, or replaced with improvements for the health of the project. For Kubernetes v1.26 there are several planned: this article identifies and describes some of them, based on the information available at this mid-cycle point in the v1.26 release process, which is still ongoing and can introduce additional changes.
+
+## The Kubernetes API Removal and Deprecation process {#k8s-api-deprecation-process}
+
+The Kubernetes project has a [well-documented deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/) for features. This policy states that stable APIs may only be deprecated when a newer, stable version of that same API is available and that APIs have a minimum lifetime for each stability level. A deprecated API is one that has been marked for removal in a future Kubernetes release; it will continue to function until removal (at least one year from the deprecation), but usage will result in a warning being displayed. Removed APIs are no longer available in the current version, at which point you must migrate to using the replacement.
+
+* Generally available (GA) or stable API versions may be marked as deprecated but must not be removed within a major version of Kubernetes.
+* Beta or pre-release API versions must be supported for 3 releases after deprecation.
+* Alpha or experimental API versions may be removed in any release without prior deprecation notice.
+
+Whether an API is removed as a result of a feature graduating from beta to stable or because that API simply did not succeed, all removals comply with this deprecation policy. Whenever an API is removed, migration options are communicated in the documentation.
+
+
+## A note about Dynamic Kubelet configuration {#dynamic-kubelet-removal}
```

Comment:
why do we highlight this separately?

Reply:
It's the only deprecation that is actually directly tracked as an enhancement (i.e. the only one that appears in the tracking board), so I used a similar approach to previous deprecation blogs. Apart from that,  there isn't anything that I can see, it's not particularly impactful so I wouldn't object to moving it to a subheading.

Reply:
Ok, +1 moving it. It might be worthwhile to discuss in the retro what "characteristics" a deprecation must have to be part of a KEP. (seems very soft and perhaps unclear?)

Reply:
This should be mentioned in the end. The only thing that was removed comparing to previous announcement: https://kubernetes.io/blog/2022/04/07/upcoming-changes-in-kubernetes-1-24/#api-removals-deprecations-and-other-changes-for-kubernetes-1-24 is the ability for new control plane to work with older kubelets. The feature was not working on kubelet for a long time now. So this only should be mentioned in the end as not important or not mentioned at all.

Reply:
Moving it to the regular section and using the distinct intro to highlight the CRI deprecation.